src/Tribe/Cost_Utils.php

```diff
@@ -0,0 +1,127 @@
+<?php
+/**
+ * Cost utility functions
+ */
+
+// Don't load directly
+if ( ! defined( 'ABSPATH' ) ) {
+	die( '-1' );
+}
+
+class Tribe__Events__Cost_Utils {
+	/**
+	 * Static Singleton Factory Method
+	 *
+	 *@return Tribe__Events__Cost_Helpers
+	 */
+	public static function instance() {
+		static $instance;
+
+		if ( ! $instance ) {
+			$className = __CLASS__;
+			$instance = new $className;
+		}
+
+		return $instance;
+	}//end instance
+
+	/**
+	 * fetches all event costs from the database
+	 *
+	 * @return array
+	 */
+	public function get_all_costs() {
+		global $wpdb;
+
+		$costs = $wpdb->get_col( "SELECT meta_value FROM {$wpdb->postmeta} WHERE meta_key = '_EventCost'" );
```

Comment:
This isn't fetching the value for a single post, instead it is fetching it for _all_ posts.